ui/src/app/stores/workflows/workflows.reducers.ts

```diff
@@ -16,27 +16,141 @@
 import * as WorkflowsActions from "../workflows/workflows.actions";
 import {ProjectModel} from "../../models/project.model";
 import {WorkflowModel} from "../../models/workflow.model";
+import {WorkflowJoinedModel} from "../../models/workflowJoined.model";
+import {WorkflowFormPartsModel} from "../../models/workflowFormParts.model";
 
 export interface State {
   projects: ProjectModel[],
-  workflows: WorkflowModel[]
-  loading: boolean
+  workflows: WorkflowModel[],
+  loading: boolean,
+  workflowAction: {
+    id: number,
+    mode: string,
+    loading: boolean,
+    workflow: WorkflowJoinedModel,
+    workflowData: {
+      details: {property: string, value: any}[],
```

Comment:
No, I dont want to do it at least here. Reducer should not involve any models. It is easier to create empty state. In case of WorkflowJoinedModel there is no other way but for data I there is.

Reply:
Hmm ok

Reply:
My concern is that when `WorkflowEntryModel` changes, you will have to change the type of `details` as well. But your concern of not involving any models is also valid. So I leave it up to you